src/smc-webapp/projects.cjsx

```diff
@@ -242,7 +242,7 @@ class ProjectsActions extends Actions
         opts = defaults opts,
             exclusions : undefined # Array<String> of sub-trees' root paths to omit
         # WARNING: Do not change the store except in a callback below.
-        block = "_fetch_directory_tree_#{project_id}"
+        block = "_fetch_directory_tree_#{project_id}_#{opts.exclusions?.toString()}"
```

Comment:
@johnjeng  -- this makes me nervous...

However, I don't care, since the whole directory tree will be sync'd via the db at some point in the future, which will remove all this.